Orleans/src/Microsoft.Orleans.Host.Abstractions/StorageProvidersFeature.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Microsoft.Orleans.Host.Abstractions
+{
+    // this would go into the (optional) storage providers DLL.
+    public interface IStorageProvidersBuilder
+    {
+        ISiloBuilder Silo { get; }
+        void AddProvider(IStorageProvider provider);
+    }
+
+    public static class SiloStorageProviderExtensions
+    {
+        public static void AddStorageProviders(this IServiceCollection services)
+        {
+            services.AddSingleton<IStorageProvidersBuilder, StorageProvidersBuilder>();
+        }
+        public static void ConfigureStorageProviders(this ISiloBuilder silo, Action<IStorageProvidersBuilder> configureMethod)
+        {
+            var factory = silo.ApplicationServices.GetRequiredService<IStorageProvidersBuilder>();
+            configureMethod(factory);
```

Comment:
Ah I see why you give `IStorageProviderBuilder` a `SiloBuilder` memeber. You want to get the registered `IStorageProvidersBuilder `from it. But this is the sevice locator pattern we want to avoid. Well I think you can do `ConfigureStorageProviders(IStorageProviderBuilder builder, Action<IStorageProvidersBuilder> configureMethod) ` and get `IStorageProvidersBuilder` before calling this method